src/components/file-browser/FilesInDirectory.test.ts

```diff
@@ -198,4 +198,40 @@ describe('FilesInDirectory', () => {
       expect(loadAllRemainingItems).toHaveBeenCalled()
     })
   })
+
+  it('uploads files', async () => {
+    // Arrange
+    const user = userEvent.setup()
+
+    const uploadFileToDirectory = jest.fn(() => Promise.resolve())
+    const mockProvider = { uploadFileToDirectory } as Partial<FileBrowserProvider> as FileBrowserProvider
```

Comment:
Why do we need both `as …` above?

Reply:
Since this mock provider doesn't implement all of FileBrowserProvider's methods, TS complains about missing properties when trying to cast directly to FileBrowserProvider.